hazelcast-client/src/main/java/com/hazelcast/client/impl/clientside/HazelcastClientInstanceImpl.java

```diff
@@ -470,6 +476,41 @@ private Credentials initCredentials(ClientConfig config) {
         return c;
     }
 
+    private void validateSecurityConfig(ClientSecurityConfig securityConfig) {
+        boolean configuredViaCredentials = securityConfig.getCredentials() != null
+                || securityConfig.getCredentialsClassname() != null;
+
+        CredentialsFactoryConfig factoryConfig = securityConfig.getCredentialsFactoryConfig();
+        boolean configuredViaCredentialsFactory = factoryConfig.getClassName() != null
+                || factoryConfig.getImplementation() != null;
+
+        if (configuredViaCredentials && configuredViaCredentialsFactory) {
+            throw new IllegalStateException("Ambiguous Credentials config. Set only one of Credentials or ICredentialsFactory");
+        }
+    }
+
+    private Credentials getCredentialsFromFactory(ClientConfig config) {
+        CredentialsFactoryConfig credentialsFactoryConfig = config.getSecurityConfig().getCredentialsFactoryConfig();
+        ICredentialsFactory factory = credentialsFactoryConfig.getImplementation();
+        if (factory == null) {
+            String factoryClassName = credentialsFactoryConfig.getClassName();
+            if (factoryClassName != null) {
+                try {
+                    factory = ClassLoaderUtil.newInstance(config.getClassLoader(), factoryClassName);
+                } catch (Exception e) {
+                    throw rethrow(e);
+                }
+            }
+        }
+        if (factory == null) {
+            return null;
+        }
+        factory.configure(config.getGroupConfig(), credentialsFactoryConfig.getProperties());
+        Credentials credentials = factory.newCredentials();
+        factory.destroy();
```

Comment:
I think this should be similar to member side lifecycle.

Reply:
This is discussed with Mehmet. He says it looks ok for now.

Reply:
Converting this behave same as member side. 
I will create new instance for every authentication request. CredentialsFactory.destroy will be called when client is shutting down.